interface/modules/custom_modules/oe-module-weno/src/Services/file_download.php

```diff
@@ -0,0 +1,198 @@
+<?php
+// Disable PHP timeout
+@ini_set('max_execution_time', '0');
+
+require_once dirname(__DIR__, 5) . "/globals.php";
+
+use OpenEMR\Common\Logging\EventAuditLogger;
+
+use OpenEMR\Common\Crypto\CryptoGen;
+use OpenEMR\Modules\WenoModule\Services\LogDataInsert;
+use OpenEMR\Modules\WenoModule\Services\WenoLogService;
+use OpenEMR\Modules\WenoModule\Services\PharmacyService;
+
+$cryptogen = new CryptoGen();
+$weno_username = $GLOBALS['weno_admin_username'];
+$weno_password = $cryptoGen->decryptStandard($GLOBALS['weno_admin_password']);
+$encryption_key = $cryptoGen->decryptStandard($GLOBALS['weno_encryption_key']);
+$baseurl = "https://online.wenoexchange.com/en/EPCS/DownloadPharmacyDirectory";
+
+$data = array(
+    "UserEmail"             => $weno_username,
+    "MD5Password"           => md5($weno_password),
+    "ExcludeNonWenoTest"    => "N",
+    "Daily"                 => "N"
+);
+
+if(date("l") == "Monday"){ //if today is Monday download the weekly file
+    $data["Daily"] = "N"; 
+}
+
+//check if there is history of download, if not do a weekly file
+// $pharmacyService = new PharmacyService();
+// $db_exist = $pharmacyService->checkWenoPharmacyLog();
+// if($db_exist != "empty"){
+//     $data["Daily"] = "Y";
+// }
+
+
+$json_object = json_encode($data);
+$method = 'aes-256-cbc';
+
+$key = substr(hash('sha256', $encryption_key, true), 0, 32);
+
+$iv = chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0) . chr(0x0);
+
+$encrypted = base64_encode(openssl_encrypt($json_object, $method, $key, OPENSSL_RAW_DATA, $iv));
+
+$fileUrl = $baseurl . "?useremail=" . $weno_username . "&data=" . urlencode($encrypted);
+$storelocation = $GLOBALS['OE_SITE_DIR'] . "/documents/logs_and_misc/weno_pharmacy.zip";
+$path_to_extract = $GLOBALS['OE_SITE_DIR'] . "/documents/logs_and_misc/";
```

Comment:
rec making another directory in /documents/logs_and_misc/, for example, /documents/logs_and_misc/weno to ensure no weird things happen if other modules/features utilize logs_and_misc directory